src/app/cic/controllers/checkDetails.js

```diff
@@ -57,20 +57,57 @@ class CheckDetailsController extends DateController {
       const middleName = req.sessionModel.get("middleName");
       const surname = req.sessionModel.get("surname")
       const fullName = firstName + " " + middleName + " " + surname
-      
+
       locals.formattedBirthDate = formatDate(dateOfBirth, "YYYY-MM-DD");
       locals.formattedExpiryDate = formatDate(expiryDate, "YYYY-MM-DD");
       locals.idChoice = idChoice;
       locals.changeUrl = `/${changeUrl}`;
       locals.fullName = fullName
 
+      if(locals.formattedExpiryDate){
+        req.sessionModel.set("expiryDate", expiryDate);
+      }
+
       callback(err, locals);
     });
   }
 
   next() {
     return '/done'
   }
+
+  async saveValues(req, res, callback) {
+
+    try {
+      const fullNameVal = req.sessionModel.get("middleName")? req.sessionModel.get("firstName") + " "+ req.sessionModel.get("middleName") + " "+ req.sessionModel.get("surname"):
+        req.sessionModel.get("firstName") +  " "+ req.sessionModel.get("surname")
+      const cicData ={
+        fullName: `${fullNameVal}`,
+        dateOfBirth: req.sessionModel.get("dateOfBirth"),
+        documentSelected:  req.sessionModel.get("photoIdChoice"),
+        dateOfExpiry: req.sessionModel.get("expiryDate")
+      }
+      await this.saveCicData(req.axios, cicData, req);
+      callback();
+
+    } catch (err) {
+      callback(err);
+    }
+
+  }
+
+  async saveCicData(axios, cicData, req) {
+
+    const headers = {
+      "session-id": req.session.tokenId,
```

Comment:
We should pick one of these and make that the header that the API uses. These two were both present in the common CRI interface for tech debt reasons. 

I recommend separating using the dash and just documenting that as our standard from now on in.

Reply:
I did try keeping just the dash one earlier but it doesn't seem to work well, will catch up tomorrow to demo it